tensorboard/plugins/hparams/api.py

```diff
@@ -448,19 +452,25 @@ def __init__(
     """
     self._hparams = _normalize_hparams(hparams)
     self._group_name = group_name if group_name is not None else ""
-    self._writer = tf.compat.v2.summary.create_file_writer(logdir)
+    if writer is None:
+      raise TypeError("writer must be a `SummaryWriter` or `str`, not None")
+    elif isinstance(writer, str):
+      self._writer = tf.compat.v2.summary.create_file_writer(writer)
+    else:
+      self._writer = writer
 
   def _write_summary(self, pb, step=None):
+    if self._writer is None:
+      raise RuntimeError(
+          "hparams Keras callback cannot be reused across training sessions"
+      )
     raw_pb = pb.SerializeToString()
-    with self._writer.as_default():
-      result = tf.compat.v2.summary.experimental.write_raw_pb(raw_pb, step=step)
-    sentinel = object()
-    if getattr(result, "numpy", sentinel) is sentinel:
-      # Not an eager tensor, so the summary wasn't actually written. We
-      # don't have access to any session context to force the op. Bail.
+    if not tf.executing_eagerly():
```

Comment:
nit: might as well move this above `pb.SerializeToString()` and not serialize the PB if we're only going to fail anyway

Reply:
Done.